lib/stdlib/src/lists.erl

```diff
@@ -102,8 +102,7 @@ equal to `Key`. Returns `Tuple` if such a tuple is found, otherwise `false`.
 -spec keyfind(Key, N, TupleList) -> Tuple | false when
       Key :: term(),
       N :: pos_integer(),
-      TupleList :: [Tuple],
-      Tuple :: tuple().
+      TupleList :: [Tuple].
```

Comment:
Doesn't Kiko's comment apply to this too? (And throughout).
If you're not going to demand tuples (technically correct, but really confusing), then don't call it a TupleList

Reply:
Also, what is this really saying... "the result Tuple must have been part of the original list"? Kind of odd (borderline unhelpful?) losing track of that the result will always be a tuple.

Reply:
I've updated the aliases and the corresponding docs to remove the mention of `TupleList`, since it is not actually required to be all tuples.

Yes this states that the result is part of the original list, which IMO conveys more information than just saying it is a `tuple()`.
If your input is `[{atom(), atom()}]` then your output will be `{atom(), atom()}`, which is more precise than just `tuple()`.